### PR TITLE
Cloudkms init

### PR DESCRIPTION
**Pull request by maraino**
### Description
This PR adds a new tool that allows initializing a PKI using Cloud KMS.

This tool is currently independent, but at some point, it should be integrated into `step ca init`

### CODE REVIEW

**Comment by dopey** [comment]
# [Codecov](https://codecov.io/gh/smallstep/certificates/pull/192?src=pr&el=h1) Report
> Merging [#192](https://codecov.io/gh/smallstep/certificates/pull/192?src=pr&el=desc) into [master](https://codecov.io/gh/smallstep/certificates/commit/8e882faf44f89178e769cf53923b4db8222bea54&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/smallstep/certificates/pull/192?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #192   +/-   ##
=======================================
  Coverage   73.85%   73.85%           
=======================================
  Files          75       75           
  Lines        8501     8501           
=======================================
  Hits         6278     6278           
  Misses       1899     1899           
  Partials      324      324           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/smallstep/certificates/pull/192?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/smallstep/certificates/pull/192?src=pr&el=footer). Last update [8e882fa...55e661b](https://codecov.io/gh/smallstep/certificates/pull/192?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: docs/kms.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    },
+}
+```
+
+Currently [step](https://github.com/smallstep/cli) does not provide an automatic
+way to initialize the public key infrastructure (PKI) using Cloud KMS, but an
+experimental tool named `step-cloudkms-init` is available for this use case. At
+some point this tool will be integrated into `step` and it will be deleted.
+
+The use `step-cloudkms-init` just enable Cloud KMS and run:
```
Comment (dopey) [inline docs/kms.md]: To use `step-cloudkms-init` just enable Cloud KMS in your `step-ca` config and run:

Comment (maraino) [reply docs/kms.md]: Fixed in 8604c31

Comment (maraino) [reply docs/kms.md]: Actually this 334d191, we don't need to enable it in the ca config.

File: cmd/step-cloudkms-init/main.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+	intermediate := &x509.Certificate{
+		IsCA:                  true,
+		NotBefore:             now,
+		NotAfter:              now.Add(time.Hour * 24 * 365 * 10),
+		KeyUsage:              x509.KeyUsageCertSign | x509.KeyUsageCRLSign,
+		BasicConstraintsValid: true,
+		MaxPathLen:            0,
+		MaxPathLenZero:        true,
+		Issuer:                pkix.Name{CommonName: "Smallstep Root"},
```
Comment (dopey) [inline cmd/step-cloudkms-init/main.go]: Should probably just reference the root.Subject here since you've already got the root certificate parsed.

Comment (maraino) [reply cmd/step-cloudkms-init/main.go]: Fixed in 6b01128

**Pull request merged**